File: app/src/main/java/ericdiaz/program/geniusplazacodingchallenge/repository/UsersRepository.java
package ericdiaz.program.geniusplazacodingchallenge.repository;

import androidx.annotation.NonNull;

import javax.inject.Inject;

import ericdiaz.program.geniusplazacodingchallenge.model.NewUser;
import ericdiaz.program.geniusplazacodingchallenge.model.UsersResponse;
import ericdiaz.program.geniusplazacodingchallenge.network.UserService;
import io.reactivex.Flowable;
import io.reactivex.Single;

/**
 * A repository for conducting network call to User API
 * <p>
 * Created 8/26/19
 *
 * @author Eric Diaz
 */

public final class UsersRepository {

    @Inject
    UserService userService;

    UsersRepository(@NonNull final UserService userService) {
        this.userService = userService;
    }

    public Flowable<UsersResponse> getUsers(final int pageNumber) {

        return userService.getUsers(pageNumber);
    }

    public Single<NewUser> createUser(@NonNull final String email,
                                      @NonNull final String firstName,
                                      @NonNull final String lastName) {

        return userService.createUser(email, firstName, lastName);
    }
}

File: app/src/main/java/ericdiaz/program/geniusplazacodingchallenge/repository/UsersRepository.kt
package ericdiaz.program.geniusplazacodingchallenge.repository

import ericdiaz.program.geniusplazacodingchallenge.model.NewUser
import ericdiaz.program.geniusplazacodingchallenge.model.UsersResponse
import ericdiaz.program.geniusplazacodingchallenge.network.UserService
import io.reactivex.Flowable
import io.reactivex.Single
import javax.inject.Inject

/**
 * A repository for conducting network call to User API
 *
 * Created 8/26/19
 *
 * @author Eric Diaz
 */

class UsersRepository(@Inject val userService: UserService) {

    fun getUsers(pageNumber: Int): Flowable<UsersResponse> {

        return userService.getUsers(pageNumber)
    }

    fun createUser(email: String,
                   firstName: String,
                   lastName: String): Single<NewUser> {

        return userService.createUser(email, firstName, lastName)
    }
}
